Modules/mmapmodule.c

```diff
@@ -682,6 +682,65 @@ mmap__exit__method(PyObject *self, PyObject *args)
     return _PyObject_CallMethodId(self, &PyId_close, NULL);
 }
 
+static PyObject *
+mmap__repr__method(PyObject *self)
+{
+    PyObject *repr;
+    mmap_object *m_obj;
+    m_obj = (mmap_object *) self;
+
+    const char *reprfmt;
+    const char *access_str;
+
+    switch (m_obj->access) {
+        case ACCESS_DEFAULT:
+            access_str = "ACCESS_DEFAULT";
+            break;
+
+        case ACCESS_READ:
+            access_str = "ACCESS_READ";
+            break;
+
+        case ACCESS_WRITE:
+            access_str = "ACCESS_WRITE";
+            break;
+
+        case ACCESS_COPY:
+            access_str = "ACCESS_COPY";
+            break;
+
+        default:
+            Py_UNREACHABLE();
+    }
+
+    Py_ssize_t size = m_obj->size;
+    const char *tp_name = Py_TYPE(self)->tp_name;
+
+
+#ifdef MS_WINDOWS
+    int fileno = m_obj->file_handle;
```

Comment:
This doesn't seem right. `HANDLE` on windows is a pointer. There is function to convert `HANDLE` to `int` but does a UNIX style fileno on WINDOWS make sense?

Reply:
In fact, `fileno` is quite inaccurate here. However I've already found a use case that giving `fileno` here as an integer will make sense, say, people might try to parse some repr string by regex, if we remove `fileno` here, it can cause some incompatibility between Unix and Windows.

What deserves to be mentioned is, there's no way to get `fileno` from `mmap` regardless of we're in Windows or Unix, so I think it'll be appealing to remove `fileno` here(although the author of this bpo-34953 has requested it and `fileno` in Unix does have useful semantics).